Sources/Datadog/Core/System/MobileDevice.swift

```diff
@@ -111,33 +111,41 @@ internal class MobileDevice {
 /// Observes "Low Power Mode" setting changes and provides `isLowPowerModeEnabled` value in a thread-safe manner.
 ///
 /// Note: this was added in https://github.com/DataDog/dd-sdk-ios/issues/609 where `ProcessInfo.isLowPowerModeEnabled` was considered
-/// not thread-safe on iOS 15. With this monitor, we change from pulling to push model for reading this property. Now, it will never be read simultaneously
-/// by multiple SDK threads - instead it will be read only once after LPM setting change and bridged to other threads through thread-safe `ValuePublisher`.
-///
-/// This should mitigate the crash originating in our SDK. We can't however prevent other code (e.g. application code) from reading this value simultaneously
-/// and causing a deadlock with SDK reads - ref. radar raised with Apple: FB9661108.
+/// not thread-safe on iOS 15. We suspect a bug present in iOS 15, where accessing `processInfo.isLowPowerModeEnabled` within a pending
+/// `.NSProcessInfoPowerStateDidChange` completion handler can sometimes lead to `_os_unfair_lock_recursive_abort` crash. The issue
+/// was reported to Apple, ref.: https://openradar.appspot.com/FB9741207
 private final class LowPowerModeMonitor {
     var isLowPowerModeEnabled: Bool {
         publisher.currentValue
     }
 
     private let publisher: ValuePublisher<Bool>
     private let notificationCenter: NotificationCenter
+    private let notificationQueue: OperationQueue
     private var powerStateDidChangeObserver: Any?
 
     init(initialProcessInfo: ProcessInfo, notificationCenter: NotificationCenter) {
         self.publisher = ValuePublisher(initialValue: initialProcessInfo.isLowPowerModeEnabled)
         self.notificationCenter = notificationCenter
+        self.notificationQueue = .main
         self.powerStateDidChangeObserver = notificationCenter
             .addObserver(
                 forName: .NSProcessInfoPowerStateDidChange,
                 object: nil,
-                queue: .main
+                queue: notificationQueue
             ) { [weak self] notification in
                 guard let processInfo = notification.object as? ProcessInfo else {
                     return
                 }
-                self?.publisher.publishAsync(processInfo.isLowPowerModeEnabled)
+
+                // We suspect an iOS 15 bug (ref.: https://openradar.appspot.com/FB9741207) which leads to rare
+                // `_os_unfair_lock_recursive_abort` crash when `processInfo.isLowPowerModeEnabled` is accessed
+                // directly in the notification handler. As a workaround, we defer its access to the next run loop
+                // where underlying lock should be already released.
+                self?.notificationQueue.addOperation {
```

Comment:
Because `WebCore` seems to run its own notification handler in reported backtraces, I was considering deferring this call even more (e.g. to 3 run loops later), just for sanity. WDYT @buranmert @maxep?

e.g.:
```
13   WebCore                                  0x190a48118     -[WebLowPowerModeObserver initWithNotifier:] + 95
14   WebCore                                  0x1914c416c     WebCore::LowPowerModeNotifier::LowPowerModeNotifier(WTF::Function<void (bool)>&&) + 51
15   WebCore                                  0x1923e9c4c     WebCore::Page::Page(WebCore::PageConfiguration&&) + 1847
```

Reply:
🤔 The idea would be to make sure that we don't read the value in the same loop as the `WebLowPowerModeObserver`? Deferring one loop is enough IMO, by reading [this](https://opensource.apple.com/source/WebCore/WebCore-7607.1.40.1.4/platform/ios/LowPowerModeNotifierIOS.mm.auto.html):
```objective-c
- (WebLowPowerModeObserver *)initWithNotifier:(WebCore::LowPowerModeNotifier&)notifier
{
    self = [super init];
    if (!self)
        return nil;

    _notifier = &notifier;
    _isLowPowerModeEnabled = [NSProcessInfo processInfo].lowPowerModeEnabled;
    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(_didReceiveLowPowerModeChange) name:NSProcessInfoPowerStateDidChangeNotification object:nil];
    return self;
}

...

- (void)_didReceiveLowPowerModeChange
{
    _isLowPowerModeEnabled = [NSProcessInfo processInfo].lowPowerModeEnabled;
    // We need to make sure we notify the client on the main thread.
    callOnMainThread([self, protectedSelf = RetainPtr<WebLowPowerModeObserver>(self)] {
        if (_notifier)
            notifyLowPowerModeChanged(*_notifier, _isLowPowerModeEnabled);
    });
}
```

It looks like `[NSProcessInfo processInfo].lowPowerModeEnabled` is invoked on the same loop as the observer. That being said we have no way of knowing if that's the actual code running on iOS, but I would be surprised!

Reply:
Sounds reasonable 👍👌, let's stick to only one runloop delay.